### ISSUE TITLE
Error: TypeError: primaryKeys is undefined

### ISSUE THREAD

**Issue by Revarh** [marmelab/ra-supabase#37]
Hi everyone,

I am facing an issue using React Admin with ra-supabase. The following message appears in the logs and I can't find why:

`TypeError: primaryKeys is undefined`

Here are my codes:

**supabase.js**
```
import { createClient } from '@supabase/supabase-js';

export const supabaseClient = createClient(
  process.env.REACT_APP_SUPABASE_URL,
  process.env.REACT_APP_SUPABASE_ANON_KEY
)
```

**baseDataProvider.js**
```
import { supabaseDataProvider } from 'ra-supabase';
import { supabaseClient } from './supabase';

const baseDataProvider = supabaseDataProvider({
    instanceUrl: process.env.REACT_APP_SUPABASE_URL,
    apiKey: process.env.REACT_APP_SUPABASE_ANON_KEY,
    supabaseClient
});

export default baseDataProvider;
```

**dataProvider.js**
```
import baseDataProvider from './baseDataProvider';
import {supabaseClient} from './supabase';

const dataProvider = {
  ...baseDataProvider,
  getList: async (resource, params) => {
    let result = null
    result = await baseDataProvider.getList(resource, params)
    return result
  },
}
```

**App.js**
```
import { Admin, CustomRoutes, ListGuesser } from 'react-admin';
import { LoginPage, SetPasswordPage, ForgotPasswordPage } from 'ra-supabase';
import { BrowserRouter, Route } from 'react-router-dom'
import { Resource } from '@react-admin/ra-rbac';

import dataProvider from './Database/dataProvider';

export const App = () => {
  const classes = CustomStyles();

  return (
    <BrowserRouter>
      <Admin
        dataProvider={dataProvider}
      >
        <CustomRoutes noLayout>
            <Route
                path={SetPasswordPage.path}
                element={<SetPasswordPage />}
            />
            <Route
                path={ForgotPasswordPage.path}
                element={<ForgotPasswordPage />}
            />
        </CustomRoutes>
        
        <Resource name="test" list={ListGuesser} />
      </Admin>
    </BrowserRouter>
  )
}
```

**Finally, in supabase:**
```
create table
  public.test (
    id bigint generated by default as identity,
    created_at timestamp with time zone not null default now(),
    first_name text null,
    last_name text null,
    constraint test_pkey primary key (id)
  ) tablespace pg_default;
```

Thank you in advance for your answers !

Best, R.

**Comment by djhi** [comment]
Thanks for reporting this. Please provide a sample application showing the issue.

**Comment by Revarh** [comment]
Hi,
Here is a test repo with the error: https://github.com/Revarh/test-ra-error
Thank you

**Comment by Revarh** [comment]
Hi,
In the meantime, trying to eliminate leads, I managed to find the issue: one package was either corrupted, or too new. Let me explain:

The project I am working on is a copy of one of a colleague, which totally works using the same code. Basically, I understood that supabaseAuthProvider() gives basic methods GetList, GetOne, etc... But it was not working with me. Either I had to create my own methods, or something was wrong. So I was misleaded for a while.

Finally I updated my packages to match those of my colleague, and only ONE was different: "@raphiniert/ra-data-postgrest" which I had in version "^2.0.0-alpha.3", and my colleague "^1.20".

Now everything is working properly.

I hope this will help if anyone gets the same error.

Best, R.
